### PR TITLE
ProjectionTestkit for Java DSL

### PR DESCRIPTION
**Pull request by renatocaval**
References #83

### CODE REVIEW

**Comment by renatocaval** [comment]
@patriknw, I'm taking a stab on this one. I didn't tested or exercise it. Will continue next week.

File: akka-projection-testkit/src/test/java/akka/projection/testkit/javadsl/ProjectionTestKitTest.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        }
+
+        @Override
+        public void run(ClassicActorSystemProvider systemProvider) {
+            CompletionStage<Done> done = mappedSource(systemProvider).asJava().runWith(Sink.ignore(), systemProvider);
+            promiseToStop.completeWith(FutureConverters.toScala(done));
+        }
+
+        @Override
+        public Future<Done> stop(ExecutionContext ec) {
```
Comment (renatocaval) [inline akka-projection-testkit/src/test/java/akka/projection/testkit/javadsl/ProjectionTestKitTest.java:206]: `Projection.stop` is returning Scala Future, but I believe we will remove the start and stop methods as they are not really needed at API level. So for the moment I'm leaving it as is. 

We may need to have a `scaladsl.Projection` and a `javadsl.Projection` for the `mappedSource`, but this still need to be explored on a future PR.

**Comment by renatocaval** [comment]
Got some free time today and did some progress on it. There is a test now written in Java to exercise the javadsl testkit.

File: akka-projection-core/src/main/scala/akka/projection/ProjectionId.scala [hunk 2]
```diff
@@ -43,6 +61,24 @@ object ProjectionId {
...
+   * For example, a "user-view" could be the name of a projection.
+   *
+   * The key must be unique for a projection name.
+   * For example, a "user-view" could have multiple projections with different keys representing different partitions,
+   * shards, etc.
+   *
+   * @param name - the projection name
+   * @param keys  - the Set of keys to associated with the passed name.
+   */
+  def of(name: String, keys: immutable.Set[String]): immutable.Set[ProjectionId] =
```
Comment (patriknw) [inline akka-projection-core/src/main/scala/akka/projection/ProjectionId.scala:80]: shouldn't that be `java.util.Set`?

Comment (renatocaval) [reply akka-projection-core/src/main/scala/akka/projection/ProjectionId.scala:80]: sure

**Comment by renatocaval** [comment]
I'm merging this one. Will add docs on a separated PR.

**Pull request merged**